src/connection.js

```diff
@@ -353,6 +371,29 @@ class Connection extends EventEmitter {
       if (config.options.useUTC != undefined) {
         this.config.options.useUTC = config.options.useUTC;
       }
+
+      // Whenever authentication property is used to specify an authentication method,
+      // the client will request encryption (the default value of the Encrypt property will be true)
+      // and it will validate the server certificate (regardless of the encryption setting), unless TrustServerCertificate = true
+      if (config.options.authentication != undefined) {
+        // check for valid options
+        if (!(config.options.authentication.toUpperCase() === this.fedAuthInfo.ValidFedAuthEnum.SqlPassword ||
+        config.options.authentication.toUpperCase() === this.fedAuthInfo.ValidFedAuthEnum.ActiveDirectoryPassword)) {
+          throw new Error('An invalid authentication method is specified');
+        }
+        if (this.config.options.tdsVersion < '7_4') {
+          throw new Error(`Azure Active Directory authentication is not supported in the TDS version ${this.config.options.tdsVersion}`);
+        }
+        this.config.options.encrypt = true;
+        this.fedAuthInfo.method = config.options.authentication;
```

Comment:
Just an FYI, `encrypt` will be true by default once #606 is merged.

Reply:
Isn't better to drop it after the PR is merged? cause without encryption fed auth would fail, so I think it's better to set it to true for now